tt_eager/tt_lib/csrc/tt_lib_bindings_tensor_dm_ops.cpp

```diff
@@ -516,6 +517,11 @@ namespace tt::tt_metal::detail{
             py::arg("batch_id"), py::arg("input_a"), py::arg("input_b"), py::arg("output_mem_config").noconvert() = operation::DEFAULT_OUTPUT_MEMORY_CONFIG, py::arg("dim") = 0,
             R"doc(Replaces batch of input in input_b denoted by batch_ids into input_a)doc"
         );
+        m_tensor.def("non_zero", &non_zero_indices,
+            py::arg("input"), py::arg("output_mem_config").noconvert() = operation::DEFAULT_OUTPUT_MEMORY_CONFIG,
+            R"doc(Returns the number of elements (N) that are non-zero as well as a tensor of the same shape as input where the first N elements are the indices of non-zero elements )doc"
+        );
+
```

Comment:
Is this function actually [https://pytorch.org/docs/stable/generated/torch.nonzero.html](https://pytorch.org/docs/stable/generated/torch.nonzero.html) ?    If so do we want to match the wording of this as well as the naming convention without the '_' ?

Reply:
hey eyon, oh I can, its not exactly a nonzero in the exact output because in PyTorch it only has a single output.
But because that would require us to shape the output based on the data of the input (not just the shape of the input) I had a compromise where I would return two outputs, one output being the number of nonzeros and a separate max-sized tensor with the nonzero indices. 

Pairing this with an unpad essentially does the same thing.